upstairs/src/volume.rs

```diff
@@ -1238,25 +1246,167 @@ impl Volume {
                 "Only a single sub_volume is supported"
             )
         }
-        // Sub volumes must all be VolumeConstructionRequest::Region
+
+        // We support replacement for read_only_parent targets with the
+        // same rules as sub_volume targets, however, for any single VCR
+        // update, only one of the sub_volume or read_only_parent targets
+        // can change.
+        let rop_result = match Self::compare_vcr_region_for_replacement(
+            o_read_only_parent,
+            n_read_only_parent,
+        ) {
+            Err(e) => {
+                warn!(log, "read_only_parent error: {}", e);
+                return Err(e);
+            }
+            Ok(rr) => rr,
+        };
+
+        // Determine the sub_volume difference if there is any.
+        let sub_volume_result = match Self::compare_vcr_region_for_replacement(
+            Some(Box::new(o_sub_volumes[0].clone())),
+            Some(Box::new(n_sub_volumes[0].clone())),
+        ) {
+            Err(e) => {
+                warn!(log, "sub_volume has invalid differences: {}", e);
+                return Err(e);
+            }
+            Ok(svr) => svr,
+        };
+
+        info!(
+            log,
+            "sub_volume_result is: {:?}, ROP:{:?}",
+            sub_volume_result,
+            rop_result
+        );
+
+        // We use the same function to compare both the region in the
+        // sub_volumes and the region in the read_only_parent.  Only one
+        // can have a different target, so we walk the cases.  As some
+        // deltas are okay for a read only parent that are not for
+        // a sub_volume, we need to look at all the possible permutations.
+        match sub_volume_result {
```

Comment:
I find tuple matches to be more clear than nested matches, e.g. `match (sub_volume_result, rop_result)`

Reply:
I made this section the tuple match, let me know if that works for you.